service/common/src/main/java/org/apache/polaris/service/catalog/PolarisCatalogHandlerWrapper.java

```diff
@@ -877,7 +883,29 @@ public LoadTableResponse loadTableWithAccessDelegation(
                   credentialDelegation.getCredentialConfig(
                       tableIdentifier, tableMetadata, actionsRequested));
             }
-            return responseBuilder.build();
+            if (accessDelegationModes.contains(AccessDelegationMode.VENDED_CREDENTIALS)) {
+              try {
+                // String hostName = InetAddress.getLocalHost().getCanonicalHostName();
+                String hostName = "localhost";
```

Comment:
This is not so simple, I'm afraid :) We need to think of proxies too.

At the very minimum, I think we need to get a `UriInfo` injected and use `uriInfo.getBaseUri()`.

A more robust approach is here: https://github.com/projectnessie/nessie/blob/a9028f12c5152ebc14ba32731b52fdc9a107db94/servers/quarkus-catalog/src/main/java/org/projectnessie/server/catalog/ExternalBaseUriImpl.java#L38